services/QuillLMS/spec/controllers/ortto_integration/webhooks_controller_spec.rb

```diff
@@ -5,30 +5,36 @@
 RSpec.describe OrttoIntegration::WebhooksController, type: :controller do
 
   describe '#create' do
-    context 'valid event payload' do
-      let!(:unsubscribed_user) {create(:user, email: 'a@b.com', send_newsletter: false)}
-      let!(:subscribed_user) {create(:user, email: 'b@c.com', send_newsletter: true)}
-
-      it 'should update the user\'s send_newsletter property to true' do
-        post :create, params: {action_name: 'subscribe', email: unsubscribed_user.email }
+    let(:ortto_secret) { 'xyz' }
 
-        expect(User.find(unsubscribed_user.id).send_newsletter).to eq true
-        expect(response.status).to eq 204
-      end
+    context 'valid event payload' do
+      let!(:subscribed_user) { create(:user, email: 'a@b.com', send_newsletter: true) }
 
       it 'should update the user\'s send_newsletter property to false' do
-        post :create, params: {action_name: 'unsubscribe', email: subscribed_user }
+        stub_const('ENV', { 'ORTTO_WEBHOOK_PASSWORD' => ortto_secret } )
 
-        expect(User.find(subscribed_user.id).send_newsletter).to eq true
-        expect(response.status).to eq 204
+        post :create, params: { email: subscribed_user.email, secret: ortto_secret }
+        expect(User.find(subscribed_user.id).send_newsletter).to eq false
+        expect(response.status).to eq 200
       end
     end
 
     context 'invalid payload' do
-      it "handles the JSON format error and reports to new relic" do
-        post :create, params: {invalid_param: 'xyz' }
+      context 'bad authentication' do
+        it 'should return 401' do
+          post :create, params: { email: 'an email', secret: 'incorrect secret' }
+          expect(response.status).to eq 403
+        end
+      end
+
+      context 'user not found' do
+        it 'should return 202 and trigger an error report' do
+          stub_const('ENV', { 'ORTTO_WEBHOOK_PASSWORD' => ortto_secret } )
+          expect(ErrorNotifier).to receive(:report)
+          post :create, params: { email: 'an email', secret: ortto_secret }
 
-        expect(response.status).to eq 400
+          expect(response.status).to eq 202
```

Comment:
I don't really have a preference on 202 or :accepted, but I think it should be consistent with the app file.